### PR TITLE
callpackage-design-pattern

### PR DESCRIPTION
**Pull request by JillThornhill**
Summarized https://nixos.org/guides/nix-pills/callpackage-design-pattern.html

### CODE REVIEW

File: maintainers/working_groups/learning_journey/documentation-survey.md [hunk 1]
```diff
@@ -167,7 +167,14 @@ To better navigate the material and judge its relevance, every entry should prov
 
 
 - https://nixos.org/guides/nix-pills/callpackage-design-pattern.html
-- https://nixos.org/guides/nix-pills/callpackage-design-pattern.html
+  
+  Introduces the callPackage pattern used extensively in `nixpkgs`. Shows how to write and use the function; use builtins.functionArgs to determine arguments, use builtins.intersectAttrs to combine attributes, and override arguments when needed.
```
Comment (fricklerhandwerk) [inline maintainers/working_groups/learning_journey/documentation-survey.md:171]:
```suggestion
  Introduces the `callPackage` pattern used extensively in Nixpkgs. Shows how to write and use the function; use `builtins.functionArgs` to determine arguments, use `builtins.intersectAttrs` to combine attributes, and override arguments when needed.
```

**Review by fricklerhandwerk — APPROVED** [review]
Thanks a lot, this is great!

Please [allow maintainers to push to your branch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests) so I can do fixups immediately and merge quickly.

**Comment by JillThornhill** [comment]
> Thanks a lot, this is great!
> 
> Please [allow maintainers to push to your branch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests) so I can do fixups immediately and merge quickly.

Ok, will do in future

**Comment by fricklerhandwerk** [comment]
Could you please commit the suggestion? Then we can merge.

**Comment by JillThornhill** [comment]
Done. Sorry!

**Pull request merged**